contracts/ModuleRegistry.sol

```diff
@@ -3,198 +3,322 @@ pragma solidity ^0.4.24;
-        delete moduleListIndex[_moduleFactory];
-
-        emit LogModuleRemoved (_moduleFactory, msg.sender);
+        // delete registry[_moduleFactory];
+        set(Encoder.getKey('registry', _moduleFactory), uint256(0));
+        // delete reputation[_moduleFactory];
+        setArray(Encoder.getKey('reputation', _moduleFactory), new address[](0));
+        // delete verified[_moduleFactory];
+        set(Encoder.getKey('verified', _moduleFactory), false);
+        // delete moduleListIndex[_moduleFactory];
+        set(Encoder.getKey('moduleListIndex', _moduleFactory), uint256(0));
+        emit ModuleRemoved(_moduleFactory, msg.sender);
         return true;
     }
 
     /**
-    * @notice Called by Polymath to verify modules for SecurityToken to use.
+    * @notice Called by Polymath to verify Module Factories for SecurityTokens to use.
     * @notice A module can not be used by an ST unless first approved/verified by Polymath
     * @notice (The only exception to this is that the author of the module is the owner of the ST)
-    * @param _moduleFactory is the address of the module factory to be registered
+    * @notice -> Only if Polymath enabled the feature.
+    * @param _moduleFactory is the address of the module factory to be verified
     * @return bool
     */
     function verifyModule(address _moduleFactory, bool _verified) external onlyOwner returns(bool) {
-        //Must already have been registered
-        require(registry[_moduleFactory] != 0, "Module factory should have been already registered");
-        verified[_moduleFactory] = _verified;
-        emit LogModuleVerified(_moduleFactory, _verified);
+        require(getUint(Encoder.getKey('registry', _moduleFactory)) != uint256(0), "Module factory must be registered");
+        set(Encoder.getKey('verified', _moduleFactory), _verified);
+        emit ModuleVerified(_moduleFactory, _verified);
         return true;
     }
 
     /**
-     * @notice Add the tag for specified Module Factory
-     * @param _moduleType Type of module.
-     * @param _tag List of tags
+     * @notice Adds a list of tags for the specified Module Factory
+     * @param _moduleType is the module type.
+     * @param _tag is the list of tags to add.
      */
     function addTagByModuleType(uint8 _moduleType, bytes32[] _tag) external onlyOwner {
          for (uint8 i = 0; i < _tag.length; i++) {
-             availableTags[_moduleType].push(_tag[i]);
+             pushArray(Encoder.getKey('availableTags', uint256(_moduleType)), _tag[i]);
          }
      }
 
     /**
-     * @notice remove the tag for specified Module Factory
-     * @param _moduleType Type of module.
-     * @param _removedTags List of tags
+     * @notice Removes the tag for specified Module Factory
+     * @param _moduleType is the module type.
+     * @param _removedTags is the list of tags to remove
      */
     function removeTagByModuleType(uint8 _moduleType, bytes32[] _removedTags) external onlyOwner {
```

Comment:
Worth noting in a `@dev` comment that this function is susceptible to a block gas limit issue if we added too many tags.